clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -898,6 +854,42 @@ private HeartbeatRequestManager createHeartbeatRequestManager(
                 heartbeatState,
                 heartbeatRequestState,
                 backgroundEventHandler,
-                metrics);
+                new Metrics());
+    }
+
+    private void mockJoiningMemberData(boolean instanceId) {
+        when(membershipManager.memberId()).thenReturn("");
+        when(membershipManager.memberEpoch()).thenReturn(0);
+        when(membershipManager.groupId()).thenReturn(DEFAULT_GROUP_ID);
+        if (instanceId)
+            when(membershipManager.groupInstanceId()).thenReturn(Optional.of(DEFAULT_GROUP_INSTANCE_ID));
+        else
+            when(membershipManager.groupInstanceId()).thenReturn(Optional.empty());
```

Comment:
what about simplifying this to:
```
        when(membershipManager.groupInstanceId()).thenReturn(Optional.ofNullable(instanceId));
```
Passing a String instanceId as param. I find it's also clearer to have the string param with the explicit `instanceId` to use (or null).

Reply:
I do like this implementation better, thank you!